configure.ac

```diff
@@ -240,7 +272,7 @@ AC_SUBST(LIBCAJA_EXTENSION_LIBS)
 
 dnl core caja
 PKG_CHECK_MODULES(GMODULE, gmodule-2.0, [GMODULE_ADD="gmodule-2.0"],[GMODULE_ADD=""])
-CORE_MODULES="glib-2.0 $GMODULE_ADD mate-desktop-2.0 gthread-2.0 gio-2.0 gio-unix-2.0 gail-3.0 libxml-2.0 $EXTRA_CORE_MODULES gtk+-3.0 x11"
+CORE_MODULES="glib-2.0 $GMODULE_ADD mate-desktop-2.0 gthread-2.0 gio-2.0 gio-unix-2.0 gail-3.0 libxml-2.0 gtk-layer-shell-0 $EXTRA_CORE_MODULES gtk+-3.0 x11"
```

Comment:
Build failure without it

Reply:
`EXTRA_CORE_MODULES` turned out to be unused, found nowhere else in the entire source directory nor in configure.ac
I found that as things stood, non-wayland builds were broken. Looks like `EXTRA_CORE_MODULES` was keeping gtk from being added to `CORE_MODULES` when gtk-layer-shell wasn't also part of it, so checked to see if anything used it, and removed it on finding no other references to it. Then had to add wayland libs in src/Makefile.am since they were no longer included in core libs. That worked

Tested three build cases after all this: 
1: no gtk-layer-shell, attempt to enable wayland. Produced the error stating that gtk-layer-shell is required for wayland
2: no gtk-layer-shell, build with --disable-wayland, successful configuration and build 
3: gtk-layer-shell installed, build with --enable-wayland, successful configuration and build